gslib/boto_resumable_upload.py

```diff
@@ -333,7 +343,11 @@ def _UploadFileBytes(self, conn, http_conn, fp, file_length,
     # in debug stream.
     http_conn.set_debuglevel(0)
     while buf:
-      http_conn.send(buf)
+      buf_bytes = buf
+      if six.PY3:
+        if isinstance(buf, str):
+          buf_bytes = buf.encode('utf-8')
+      http_conn.send(buf_bytes)
```

Comment:
Moving my comment to the "right" side of the diff (GitHub for some reason put it on the left last time):

This should be `total_bytes_uploaded += len(buf_bytes)` instead, since `buf_bytes` could be longer than `buf`, its `str` equivalent -- calling `len()` on a unicode object counts the number of code points, not the number of bytes needed to encode the thing in utf-8 or any other codec.